pvmismatch/contrib/ModuleMismatchSimulator.py

```diff
@@ -0,0 +1,510 @@
+'''
+Created on Mar 29, 2013
+This script allows the user to dynamically investigate the IV and PV
+characteristics of a single module. The user chooses the modules size--72 or 96
+cells. A UI is then generated that allows the user to change the size, location,
+and irradiance level of a single "shade rectangle". Outputs include cell,
+substring, and module level IV and PV curves as well as a module diagram showing
+the shade location, any reverse biased cells, and any active diodes.
+@author: bmeyers
+'''
+# ==============================================================================
+# Importing standard libraries
+# ==============================================================================
+
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+import numpy as np
+from scipy.interpolate import interp1d
+from matplotlib.widgets import Slider
+from matplotlib.widgets import Button
+import json
+from functools import partial
+from copy import deepcopy
+
+# ==============================================================================
+# Import PVmismatch items
+# ==============================================================================
+
+try:
+    from pvmismatch import PVsystem, PVmodule, PVcell, PVconstants
+    from pvmismatch.pvmismatch_lib.pvmodule import STD72, STD96, STD128
+except ImportError:
+    import sys
+    sys.path.append('../../')
+    from pvmismatch import PVsystem, PVmodule, PVcell, PVconstants
+    from pvmismatch.pvmismatch_lib.pvmodule import STD72, STD96, STD128
```

Comment:
Can you drop this, and just import pvmismatch? I don't want to endorse altering `sys.path` in code. This can be done from CLI using `pip install -e path/to/pvmismatch` or `export PYTHONPATH=path/to/pvmismatch:$PYTHONPATH`. Ideally `sys.path` should only be used in an interpreter, not in a script or package module. IMO runtime alterations of path are a ["code smell"][1] that should be avoided.

[1]: https://en.wikipedia.org/wiki/Code_smell

---
https://blog.codinghorror.com/code-smells/
http://wiki.c2.com/?CodeSmell
https://sourcemaking.com/refactoring/smells
https://sourcemaking.com/antipatterns
https://en.wikipedia.org/wiki/Code_smell